src/parser.ts

```diff
@@ -1,163 +1,274 @@
+  accessor = this.RULE("accessor", () => {
+    this.OR([
+      { ALT: () => this.CONSUME(Member) },
+      { ALT: () => this.CONSUME(WildcardMember) },
+      { ALT: () => this.SUBRULE(this.arrayAccessor) },
+      { ALT: () => this.CONSUME(WildcardArray) },
+      { ALT: () => this.SUBRULE(this.filterExpression) },
+      { ALT: () => this.CONSUME(ItemMethod) }
+    ])
   })
 
-  existsPathQuery = this.RULE("existsPathQuery", () => {
-    this.CONSUME(FilterValue)
+
+  arrayAccessor = this.RULE("array", () => {
+    this.CONSUME(LeftBracket)
+    this.AT_LEAST_ONE_SEP({
+      SEP: Comma,
+      DEF: () => this.SUBRULE(this.subscript)
+    })
+    this.CONSUME(RightBracket)
+  })
+
+
+  subscript = this.RULE("subscript", () => {
+    this.SUBRULE(this.wff)
     this.OPTION(() => {
-      this.CONSUME(PathSeparator)
-      this.SUBRULE(this.pathQuery)
+      this.CONSUME(To)
+      this.SUBRULE1(this.wff)
     })
   })
 
-  existsQuery = this.RULE("existsQuery", () => {
-    this.CONSUME(Exists)
-    this.CONSUME(LeftParen)
-    this.SUBRULE(this.existsPathQuery, { LABEL: "query" })
+/*
+<JSON filter expression> ::=
+          <question mark> <left paren> <JSON path predicate> <right paren>
+ */
+  filterExpression = this.RULE("filter", () => {
+    this.CONSUME(PredicateStart)
+    this.SUBRULE(this.pathPredicate)
     this.CONSUME(RightParen)
   })
 
-  conditionalOperator = this.RULE("conditionalOperator", () => {
-    this.CONSUME(ConditionalOperator, { LABEL: "operator" })
-  })
 
-  pathPart = this.RULE("pathPart", () => {
-    this.CONSUME(Identifier, { LABEL: "name" })
-    this.OPTION(() =>
-      this.OR([
-        { ALT: () => this.SUBRULE(this.arrayAccessor) },
-        { ALT: () => this.SUBRULE(this.arguments) }
-      ]))
+  delimitedPredicate = this.RULE("delPred", () => {
+    this.OR([
+      { ALT: () => this.SUBRULE(this.exists) },
+      { ALT: () => this.SUBRULE(this.scopedPredicate) }
+    ])
   })
 
-  group = this.RULE("group", () => {
-    this.CONSUME(LeftParen, { LABEL: "group" })
-    this.SUBRULE(this.wff)
-    this.CONSUME(RightParen, { LABEL: "group" })
-  })
 
-  op = this.RULE("op", () => {
-    this.OR1([
-      { ALT: () => this.CONSUME(ArithmeticOperator, { LABEL: "connector" }) },
-      { ALT: () => this.CONSUME(To, { LABEL: "connector" }) }
-    ])
-    this.OR2([
-      { ALT: () => this.SUBRULE1(this.group, { LABEL: "rhs_group" }) },
-      { ALT: () => this.CONSUME1(Integer, { LABEL: "rhs_integer" }) },
-      { ALT: () => this.CONSUME1(Last, { LABEL: "rhs_last" }) }
+  // todo more than 3 alts so cache them: https://chevrotain.io/docs/guide/performance.html#caching-arrays-of-alternatives
+  predicate = this.RULE("pred", () => {
+    this.OR([
+      // IGNORE_AMBIGUITIES because we are in a filter expression, but might not be ok in the end.
+      { ALT: () => this.SUBRULE(this.delimitedPredicate), IGNORE_AMBIGUITIES: true },
```

Comment:
If I turn off IGNORE_AMBIGUITIES then it can't match the various rules that start with `(` and can have nested `(`. I would like to remove the ambiguity, but I am not sure that is possible, or that it matters. There will be no end of `((((` statements that can be either a `scopedWff` or a `scopedPred`.